test/static/accept/ufuncs.pyi

```diff
@@ -53,3 +53,29 @@ assert_type(np.isinf(AR_f8, out=AR_bool), npt.NDArray[np.bool_])
 assert_type(np.isfinite(f8), np.bool_)
 assert_type(np.isfinite(AR_f8), npt.NDArray[np.bool_])
 assert_type(np.isfinite(AR_f8, out=AR_bool), npt.NDArray[np.bool_])
+
+assert_type(np.logical_not(True), np.bool_)
+assert_type(np.logical_not(AR_bool), npt.NDArray[np.bool_])
+assert_type(np.logical_not(AR_bool, out=AR_bool), npt.NDArray[np.bool_])
+assert_type(np.logical_not(AR_bool, dtype=np.object_), npt.NDArray[np.object_] | np.bool_)
+
+assert_type(np.logical_and(True, True), np.bool_)
+assert_type(np.logical_and(AR_bool, AR_bool), npt.NDArray[np.bool_ | np.object_])
```

Comment:
Like I said earlier; it's better to avoid returning unions if you can. This result type isn't something that users would expect to see.